src/main/java/com/matt/forgehax/mods/CoordsFinder.java

```diff
@@ -0,0 +1,191 @@
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Integer> minPlayerDist =
+      getCommandStub()
+          .builders()
+          .<Integer>newSettingBuilder()
+          .name("player-min-dist")
+          .description("how far a player teleport has to be from you to get logged")
+          .min(0)
+          .defaultTo(256)
+          .build();
+
+  private final Path log = getFileManager().getBaseResolve("logs/coordsfinder.log");
+  private final SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+  private BufferedWriter logWriter;
+
+  public CoordsFinder() {
+    super(Category.MISC, "CoordsFinder", false, "Logs coordinates of lightning strikes and teleports");
+  }
+
+  @Override
+  protected void onEnabled() {
+    super.onEnabled();
+
+    try {
+      if (!Files.exists(log)) Files.createFile(log);
+
+      logWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(log.toFile(), true)));
+    } catch (Throwable t) {
+      t.printStackTrace();
+    }
+  }
+
+  @Override
+  protected void onDisabled() {
+    try {
+      if (nonNull(logWriter))
+        logWriter.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } finally {
+      logWriter = null;
+    }
+
+    super.onDisabled();
+  }
+
+  private void logCoords(String name, double x, double y, double z) {
+    int ix = MathHelper.floor(x);
+    int iy = MathHelper.floor(y);
+    int iz = MathHelper.floor(z);
+
+    printInform("%s > [x:%d, y:%d, z:%d]", name, ix, iy, iz);
+
+    if (nonNull(logWriter)) {
+      try {
+        logWriter.write(String.format("[%s][%s][%d,%d,%d]", timeFormat.format(new Date()), name, ix, iy, iz));
+        logWriter.newLine();
+        logWriter.flush();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  private void logCoordsOnMinecraftThread(String name, double x, double y, double z) {
+    MC.addScheduledTask(() -> logCoords(name, x, y, z));
+  }
+
+  private boolean pastDistance(EntityPlayer player, BlockPos pos, double dist) {
+    return player.getDistanceSqToCenter(pos) >= Math.pow(dist, 2);
+  }
+
+  @SubscribeEvent
+  public void onPacketRecieving(PacketEvent.Incoming.Pre event) {
+    EntityPlayer player = getLocalPlayer();
+    WorldClient world = getWorld();
+    if (isNull(player) || isNull(world)) return;
+
+    if (logLightning.get() && event.getPacket() instanceof SPacketSoundEffect) {
+      SPacketSoundEffect packet = event.getPacket();
+
+      // in the SPacketSpawnGlobalEntity constructor, this is only set to 1 if it's a lightning bolt
+      if (packet.getSound() != SoundEvents.ENTITY_LIGHTNING_THUNDER) return;
+
+      BlockPos pos = new BlockPos(packet.getX(), packet.getY(), packet.getZ());
+
+      if (pastDistance(player, pos, minLightningDist.get()))
+        logCoordsOnMinecraftThread("Lightning strike", pos.getX(), pos.getY(), pos.getZ());
+    } else if (event.getPacket() instanceof SPacketEntityTeleport) {
```

Comment:
I don't think this will do the trick. You should be listening to the entity being removed from the loadedEntityList, and if its position when its removed is outside the players visible range, then it should be logged. Make sure nether portals don't accidentally trigger this.

Reply:
> if its position when its removed is outside the players visible range

And where does it get outside the visible range? With a teleport.

I teleported my main account to X 2000 and the modded player got a log in chat (and the file) telling them exactly where I went, because the wolf teleported with me. You can see this in one of the images. However, the server simply sent a delete packet to them so they didn't get a _player_ teleport notification.

Spigot suppresses these packets for players and tells the client to delete the entity instead. There is no teleport packet before that, so the player just gets removed where they were. I tested the player teleportation thing with some enderpearls because all I have is a Spigot server, and it also worked okay.

This works for wolves, but on Spigot it doesn't work for players. I haven't tested it with vanilla yet though.

Reply:
All I know is that people report to see tracers of players that pearl away from them for a brief moment. That indicates that the server sends an entity update packet detailing the players position where he teleported to. I don't know if this is the same with wolfs, but I think the same methodology would work.

Oh, also one recommendation is to log said players name alongside their coordinates.

Reply:
> the server sends an entity update packet detailing the players position where he teleported to

🤔 Sure wish there was an if statement here that logged that...

> Oh, also one recommendation is to log said players name alongside their coordinates.

Yes.

Reply:
Oh so the server does send a teleport packet? Even on spigot? If so then forget my previous comment.

Reply:
Sorry that I sound so unsure. It's because I am only going off rumors that this trick works, it might even be patched now by spigot.

Reply:
Yes it is patched by spigot as I said in my comment. The server does not update the position on the client, the server tells the client to delete the entity immediately. When the player comes back into range, the server tells the client to recreate the player. I have packet logs that confirm this.